pkg/proxy/ipvs/netlink_linux.go

```diff
@@ -133,7 +133,10 @@ func (h *netlinkHandle) ListBindAddress(devName string) ([]string, error) {
 func (h *netlinkHandle) GetAllLocalAddresses() (sets.Set[string], error) {
 	addr, err := net.InterfaceAddrs()
 	if err != nil {
-		return nil, fmt.Errorf("Could not get addresses: %v", err)
+		err = proxyutil.GetInterfaceAddrsByInterfaces(&addr)
```

Comment:
Is there a good reason for calling the bad function first and then calling the good function only as a workaround? Why don't we just ignore the stdlib function and always use our own?

Reply:
I agree with Dan, I don't like the double check, and fall back to a method we had to suppress because of the performance problem.

This file is linux specific, if we are worried about windows in the other function then we do net.InterfaceAddrs() for windows and our netlink for linux